example/mapreduce/task_interface_mapreduce.go

```diff
@@ -0,0 +1,19 @@
+package mapreduce
+
+import (
+	"github.com/taskgraph/taskgraph"
+)
+
+// MapreduceTask is a logic repersentation of a computing unit in mapreduce framework.
+// Each task contains at least one Node.
+
+type MapreduceTask interface {
+	// task interface of mapreduce framework
+	taskgraph.Task
+	// For mapper work, after processing by function of user
+	// it need provide user mapper function a method to emit their result to shuffle,
+	// Thus framework could take these data to specfic shuffle.
+	Emit(string, string)
```

Comment:
For example, user implements these two function
```
func mapperFunc(mp MapreduceTask, text string) {
	textReader := bufio.NewReader(strings.NewReader(text))
	var err error
	var str string
	for err != io.EOF {
		str, err = textReader.ReadString(' ')
		if err != io.EOF && err != nil {
			return
		}
		if err != io.EOF {
			str = str[:len(str)-1]
		}
		mp.Emit(str, "1")
	}
}

func reducerFunc(mp MapreduceTask, key string, val []string) {
	lenVal := len(val)
	var sum int
	for v := 0; v < lenVal; v++ {
		a, err := strconv.Atoi(val[v])
		if err != nil {
			continue
		}
		sum = sum + a
	}
	mp.Collect(key, strconv.Itoa(sum))
}

```

user transmit these two function to framework.
For instance, when mapper task processing a mapper work, it could invoke the mapper function user provides.